### ISSUE TITLE
Add attribution patching

### ISSUE THREAD

**Issue by atticusg** [goodfire-ai/causalab#11]
Add a "attribution_patching=False" keyword argument flag to the intervention_experiment.perform_intervention() method that approximates all of the interventions using attribution patching. 

This will fully override the typical intervention set up, and instead, for each batch of inputs:

1. Run a forward pass and use the function _collect_features in pyvene_core.py to harvest all of the features for the original inputs and counterfactual inputs, and the output logits for the original inputs, and also the gradients for the counterfactual inputs with respect to the output logits of a model. 
2. For each model_units in model_units_list, take the original features, subtract the counterfactual features, and then multiply by the gradient with respect to the logits to get the expected effect of the intervention. 
3. Add the effect of the intervention to the output logits for the original input to compute the expected logits after intervention.

_collect_features will need to be modified to collect the gradients for the counterfactual features and the output logits for the original inputs. 

This will be a bit complicated, but incredibly high impact for our efficiency
